api/src/routing.js

```diff
@@ -150,25 +152,50 @@ app.use(compression());
 
 // TODO: investigate blocking writes to _users from the outside. Reads maybe as well, though may be harder
 //       https://github.com/medic/medic/issues/4089
+const root = (req, res) => {
+  if (('deviceID' in req.query) || req.is('application/json')) {
+    // couchdb request - let it go
+    return proxy.web(req, res);
+  }
+  let redirectUrl = '/';
+  const startPosition = req.url.indexOf('#');
+  if (startPosition !== -1) {
+    redirectUrl = redirectUrl.concat(req.url.slice(startPosition));
+  }
+  res.redirect(redirectUrl);
+};
 
 app.get('/', function(req, res) {
   if (req.headers.accept === 'application/json') {
     // couchdb request - let it go
-    proxy.web(req, res);
+    return proxy.web(req, res);
   } else {
-    // redirect to the app path - redirect to _rewrite
-    res.redirect(appPrefix);
+    res.sendFile(path.join(STATIC_RESOURCE_DESTINATION, 'templates/inbox.html'));
   }
 });
+app.get(appPrefix, root);
 
-/*
-To facilitate service worker prefetch on Chrome <66, serve a version of the app which does not require authentication
-*/
-app.get(appPrefix, (req, res, next) => {
-  if ('_sw-precache' in req.query) {
-    return res.sendFile(path.join(__dirname, 'extracted-resources/templates/inbox.html'));
+app.all(/^\/(medic)\/(.?)/, (req, res, next) => {
+  if (environment.db === 'medic') {
+    return next();
+  }
+  const originalUrl = req.url;
+  req.url = `/${environment.db}${originalUrl.slice(6)}`;
+  if (originalUrl.includes('login')) {
+    res.redirect(req.url);
+  } else {
+    proxy.web(req, res);
+  }
+}); 
```

Comment:
I'm not sure about this block - I think I asked about it in a previous PR but it's still not clear to me.

Firstly the regex is more complicated than it needs to be. I think the parens are unnecessary?

Secondly, it's only for e2e testing, right? The if block at the top fires for all non-e2e requests? Why are we redirecting or proxying e2e tests from `/medic-test/*` to `/medic/*`? Won't this mean e2e tests actually end up running against the `medic` db not the `medic-test` db?

I think we can remove this whole block.

Reply:
I agree, I will simplify it by replacing the regex with `/medic/*`

The reason we did this route handler is because in our angular code we have hard coded the db name `medic` and we don't want to be rigid that the couch db must be named `medic`, a good case scenario is `medic-test`.

So this route essentially replace the `medic` with `environment.db` and makes the request. If both `medic` and `environment.db` matches I just call `next()` because actually what do we need to replace there?

e2e cannot run against the `medic` db and not `medic-test` that will be an epic fail, it is actually the main reason that led to the birth of this `/medic/*` route handler after we had hard coded db name on Angular

Reply:
Makes sense. Thanks for explaining it!